docs/admin/extensible-admission-controllers.md

```diff
@@ -0,0 +1,152 @@
+---
+assignees:
+- smarterclayton
+- lavalamp
+- whitlockjc
+- caesrxuchao
+title: Extending admission controllers
+---
+
+* TOC
+{:toc}
+
+## The plugin-style admission controllers are not flexible enough
+
+[Using Admission Controllers documentation](/doc/admin/admission-controllers.md) introduces how to use the plugin-style admission controllers.The plugin admission controllers need to be compiled in-tree, and are only configurable when the apiserver starts up.
+
+## How do I extend admission controllers?
+
+In 1.7, we introduce two alpha features, *Initializers* and *External Admission Hooks*, they allow admission controllers to be developed out-of-tree and to be configured at runtime.
+
+We will first describe initializers and then describe external admission hooks.
+
+## Initializers
+
+###what are they?
+* muntating AC; called in serial
+* a plugin-style initializer admission controller sets the default list of initializers for objects in the CREATE request. The default list of initializers can be dynamically configured via `InitializersConfiguration`, see XXX
+* an `initializer agent` is the worker that mutates the objects. is a controller watch for objects with `metadata.finalizers[0]==<initializer id>`
+
+* uninitialized objects are not visible ...
+
+### When are initializers called?
+
+## External Admission Webhooks (assigned to @whitlockjc)
+
+###What are they? (assigned to @whitlockjc)
+External webhook: non-mutating, called in parallel
+
+### When are they called? (assigned to @whitlockjc)
+
+AdmissionChain, depends on the admission-controller-config
+
+Recommended plug-in order:
+???
+```
+--admission-control=Initializer,NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota,DefaultTolerationSeconds,GenericAdmissionWebhook"
+```
+
+## How to enable initializers and external admission webhooks?
+
+Initializers and External Admission Webhooks are alpha features, which are disabled by default. To turn them on, you need to
+
+* include "Initializer" and "GenericAdmissionWebhook" in the `--admission-control` flag when starting the apiserver. If you are using HA-master or aggregated apiservers, you need to do so for each apiserver.
+
+* enable the dynamic admission controller registration API by setting `--runtime-config=admissionregistration.k8s.io/v1alpha1` when starts the kube-apiserver.
+
+## How to write an initializer (@Clayton)
+## How to deploy an initializer (@Clayton)
+## How to configure initializers on the fly?
+
+You can configure what initializers are enabled and what resources are subject to the initializers via creating initiallizerconfigurations.
```

Comment:
I intend to refer to the API `initiallizerconfigurations`, so one word the starts with a lowercase "i", as it appears in the api doc.